File: plugins/java/tube/src/dev/argon/plugins/tube/TubeOptions.java
package dev.argon.plugins.tube;

import dev.argon.plugin.api.options.*;
import java.nio.file.Path;
import java.util.List;
import org.checkerframework.checker.nullness.qual.*;

@Options(decoding = OptionsDecoding.NORMAL)
record TubeOptions<E extends Exception>(
) {
}

File: argon-plugins-tube/src/main/scala/dev/argon/plugins/tube/TubeOptions.scala
package dev.argon.plugins.tube

import dev.argon.options.OptionCodec
import dev.argon.parser.SyntaxError

import java.nio.charset.CharacterCodingException

final case class TubeOptions[-R, +E]()

object TubeOptions:

  given optionCodec[R, E >: TubeError]: OptionCodec[R, E, TubeOptions[R, E]] =
    OptionCodec.derive

end TubeOptions

